src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs

```diff
@@ -2993,20 +2993,6 @@ private void OutputStartingBrace()
             }
         }
 
-        private CompilerResults FromFileBatch(CompilerParameters options, string[] fileNames)
-        {
-            if (options == null)
-            {
-                throw new ArgumentNullException(nameof(options));
-            }
-            if (fileNames == null)
-            {
-                throw new ArgumentNullException(nameof(fileNames));
-            }
```

Comment:
These checks/exceptions should be kept.  I would suggest keeping this method, and then instead of adding FromFileBatch to another file, add FromFileBatchCore (or something similarly named), which this method can call instead of throwing.  That way any implementations of FromFileBatch will share the same error validation.